personal_finances/bank_interface/password_handler.py

```
@@ -0,0 +1,24 @@
+import argon2
+import logging
+
+USER_PASSWORD_MIN_LEN = 6
+LOGGER = logging.getLogger(__name__)
+
+
+def password_match(recv_password: str, stored_password: str) -> bool:
+    LOGGER.info("Argon2 comparisson start")
+    try:
+        argon2.PasswordHasher().verify(
+            stored_password.encode("utf-8"), recv_password.encode("utf-8")
+        )
+        return True
+    except argon2.exceptions.VerifyMismatchError:
+        return False
+    except Exception as e:
```

Let's not catch all exceptions and let them bubble up, technically we can't tell this is a password mismatch

---

Excluded this case, but included a mismatch one, that catches if it is a comparison to another hash format, as an outdated encryption library hash stored.

---

This new exception caught will cause the same exit ("User or password incorrect") if the hash format doesn't match with the stored one. This is an solution for previous users, what requires a password reset (feature needed)